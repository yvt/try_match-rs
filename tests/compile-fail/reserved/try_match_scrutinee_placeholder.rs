fn main() {
    // scrutinee placeholder is reserved for partial application
    // <https://github.com/yvt/try_match-rs/issues/3>
    let _ = try_match::try_match!(_, Some(_0))(Some(42));
}
