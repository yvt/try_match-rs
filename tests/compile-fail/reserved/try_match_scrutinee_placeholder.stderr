error: no rules expected the token `_`
 --> tests/compile-fail/reserved/try_match_scrutinee_placeholder.rs:4:35
  |
4 |     let _ = try_match::try_match!(_, Some(_0))(Some(42));
  |                                   ^ no rules expected this token in macro call
  |
note: while trying to match meta-variable `$r#in:expr`
 --> src/lib.rs
  |
  |     ($in:expr, $(|)? $($p:pat)|+ $(if $guard:expr)? => $out:expr $(,)?) => {
  |      ^^^^^^^^
